src/Core/src/Platform/iOS/WindowViewController.cs

```diff
@@ -0,0 +1,185 @@
+	{
+		UpdateContentWrapperContentFrame();
+		base.ViewDidLayoutSubviews();
+	}
+
+	void UpdateContentWrapperContentFrame()
+	{
+		// At this point the _contentWrapperView bounds haven't been set
+		// so we just use the windows bounds to set this value
+		var frame = new CGRect(0, 0, View!.Bounds.Width, View!.Bounds.Height - (_contentWrapperTopConstraint?.Constant ?? 0));
+
+		if (_contentWrapperView is not null && _contentWrapperView.Subviews[0].Frame != frame)
+		{
+			_contentWrapperView.Subviews[0].Frame = frame;
+		}
+	}
+	
+	/// <summary>
+	/// Sets up the TitleBar in the ViewController.
+	/// </summary>
+	/// <param name="window">An instance of the <see cref="IWindow"/>.</param>
+	/// <param name="mauiContext">An instance of the <see cref="IMauiContext"/>.</param>
+	public void SetUpTitleBar(IWindow window, IMauiContext mauiContext)
+	{
+		var platformWindow = window.Handler?.PlatformView as UIWindow;
+
+		if (platformWindow is null || View is null)
+		{
+			return;
+		}
+
+		var newTitleBar = window.TitleBar?.ToPlatform(mauiContext);
+
+		IView? iTitleBar = null;
+		_iTitleBarRef?.TryGetTarget(out iTitleBar);
+
+		if (newTitleBar != iTitleBar)
+		{
+			_titleBar?.RemoveFromSuperview();
+			iTitleBar?.DisconnectHandlers();
+			iTitleBar = null;
+
+			if (newTitleBar is not null)
+			{
+				iTitleBar = window.TitleBar;
+				View.AddSubview(newTitleBar);
+			}
+
+			_titleBar = newTitleBar;
+			_iTitleBarRef = new WeakReference<IView?>(iTitleBar);
+		}
+		
+		_isTitleBarVisible = (iTitleBar?.Visibility == Visibility.Visible);
+
+		var platformTitleBar = platformWindow.WindowScene?.Titlebar;
+
+		if (newTitleBar is not null && platformTitleBar is not null)
+		{
+			platformTitleBar.TitleVisibility = UITitlebarTitleVisibility.Hidden;
+		}
+
+		else if (newTitleBar is null && platformTitleBar is not null)
+		{
+			platformTitleBar.TitleVisibility = UITitlebarTitleVisibility.Visible;
+		}
+
+		IsFirstLayout = true;
+		LayoutTitleBar();
+	}
+
+	/// <summary>
+	/// Measures and arranges the TitleBar and adjusts the frame for the window content to make space for the TitleBar.
+	/// </summary>
+	public void LayoutTitleBar()
+	{
+		if (_contentWrapperTopConstraint is null || View is null)
+		{
+			return;
+		}
+
+		var current = _contentWrapperTopConstraint.Constant;
+		_iTitleBarRef.TryGetTarget(out var iTitleBar);
+
+		nfloat titleBarHeight = 0;
+
+		if (_isTitleBarVisible && iTitleBar is not null)
+		{
+			var measured = iTitleBar.Measure(View.Bounds.Width, double.PositiveInfinity);
+			iTitleBar.Arrange(new Graphics.Rect(0, 0, View.Bounds.Width, measured.Height));
+			titleBarHeight = (nfloat)measured.Height;
+		}
+
+		_contentWrapperTopConstraint.Constant = titleBarHeight;
+
+		UpdateContentWrapperContentFrame();
+	}
+
+	public void SetTitleBarVisibility(bool isVisible)
+	{
+		if (_contentWrapperTopConstraint is null || View is null)
+			return;
+
+		_isTitleBarVisible = isVisible;			
+		LayoutTitleBar();
```

Comment:
Can we just invalidate the layout on WindowViewController here instead of calling `LayoutTitleBar`?

Reply:
Yeah that looks like it works well!